fuzzer/ultrahdr_enc_fuzzer.cpp

```diff
@@ -121,34 +116,76 @@ void UltraHdrEncFuzzer::process() {
+    auto multi_channel_gainmap = mFdp.ConsumeIntegral<int8_t>();
 
     // gainmap scale factor
-    auto gm_scale_factor = mFdp.ConsumeIntegralInRange<uint8_t>(0, 150);
+    auto gm_scale_factor = mFdp.ConsumeIntegralInRange<int16_t>(-32, 192);
 
     // encoding speed preset
     auto enc_preset = mFdp.ConsumeBool() ? UHDR_USAGE_REALTIME : UHDR_USAGE_BEST_QUALITY;
 
-    // gainmap gamma
+    // gainmap metadata
+    auto minBoost = mFdp.ConsumeFloatingPointInRange<float>(-4.0f, 64.0f);
+    auto maxBoost = mFdp.ConsumeFloatingPointInRange<float>(-4.0f, 64.0f);
     auto gamma = mFdp.ConsumeFloatingPointInRange<float>(-1.0f, 5);
-
-    // content boost
-    auto minBoost = mFdp.ConsumeFloatingPointInRange<float>(-1.0f, 100);
-    auto maxBoost = mFdp.ConsumeFloatingPointInRange<float>(minBoost - 1.0f, 100);
+    auto offsetSdr = mFdp.ConsumeFloatingPointInRange<float>(-1.0f, 1.0f);
+    auto offsetHdr = mFdp.ConsumeFloatingPointInRange<float>(-1.0f, 1.0f);
+    auto minCapacity = mFdp.ConsumeFloatingPointInRange<float>(-4.0f, 48.0f);
+    auto maxCapacity = mFdp.ConsumeFloatingPointInRange<float>(-4.0f, 48.0f);
+
+    // raw buffer config
+    bool hasHdrStride = mFdp.ConsumeBool();
+    int yHdrStride = mFdp.ConsumeIntegralInRange<uint16_t>(width, width + 128);
+    if (!hasHdrStride) yHdrStride = width;
+    bool isHdrUVContiguous = mFdp.ConsumeBool();
+    bool hasHdrUVStride = mFdp.ConsumeBool();
+    int uvHdrStride = mFdp.ConsumeIntegralInRange<uint16_t>(width, width + 128);
+    if (!hasHdrUVStride) uvHdrStride = width;
+
+    bool hasSdrStride = mFdp.ConsumeBool();
+    int ySdrStride = mFdp.ConsumeIntegralInRange<uint16_t>(width, width + 128);
+    if (!hasSdrStride) ySdrStride = width;
+    bool isSdrUVContiguous = mFdp.ConsumeBool();
+    bool hasSdrUVStride = mFdp.ConsumeBool();
+    int uvSdrStride = mFdp.ConsumeIntegralInRange<uint16_t>(width / 2, width / 2 + 128);
+    if (!hasSdrUVStride) uvSdrStride = width / 2;
+
+    // editing effects
+    auto applyMirror = mFdp.ConsumeBool();
+    uhdr_mirror_direction_t direction =
+        mFdp.ConsumeBool() ? UHDR_MIRROR_VERTICAL : UHDR_MIRROR_HORIZONTAL;
+
+    auto applyRotate = mFdp.ConsumeBool();
+    int degrees = degrees = mFdp.PickValueInArray({-90, 0, 90, 180, 270});
+
+    auto applyCrop = mFdp.ConsumeBool();
+    int left = mFdp.ConsumeIntegral<int16_t>();
+    int right = mFdp.ConsumeIntegral<int16_t>();
+    int top = mFdp.ConsumeIntegral<int16_t>();
+    int bottom = mFdp.ConsumeIntegral<int16_t>();
+
+    auto applyResize = mFdp.ConsumeBool();
+    int resizeWidth = mFdp.ConsumeIntegralInRange<int16_t>(-32, kMaxWidth + 128);
+    int resizeHeight = mFdp.ConsumeIntegralInRange<int16_t>(-32, kMaxHeight + 128);
+
+    // exif
+    char greeting[] = "Exif says hello world";
+    uhdr_mem_block_t exif{greeting, mFdp.ConsumeIntegralInRange<uint8_t>(0, sizeof greeting * 2),
+                          sizeof greeting};
 
     // ALOGV("encoding configuration options : ");
```

Comment:
I think these are OK for now, but better to have a macro define that turn on/off the log.

Reply:
Done. Only if UHDR_ENABLE_LOGS are enabled the prints will be seen.